src/main/scala/com/lucidworks/spark/SolrRelation.scala

```diff
@@ -323,5 +335,43 @@ object SolrRelation {
     })
     unknownParams
   }
+
+  def isStreamingPossible(querySchema: StructType, baseSchema: StructType, query: SolrQuery): Boolean = {
+    // Check if all the fields in the querySchema have docValues enabled
+    for (structField <- querySchema.fields) {
+      val metadata = structField.metadata
+      if (!metadata.contains("docValues"))
+        return false
+      if (metadata.contains("docValues") && !metadata.getBoolean("docValues"))
+        return false
+    }
+
+    val sortClauses = query.getSorts.asScala.toList
+
+    if (sortClauses.nonEmpty) {
+      // Check if the sorted field (if exists) has docValue enabled
+      for (sortClause: SortClause <- sortClauses) {
+        val sortField = sortClause.getItem
+        if (baseSchema.contains(sortField)) {
+          val sortFieldMetadata = baseSchema(sortField).metadata
+          if (!sortFieldMetadata.contains("docValues"))
+            return false
+          if (sortFieldMetadata.contains("docValues") && !sortFieldMetadata.getBoolean("docValues"))
+            return false
+        } else {
+          log.warn("The sort field '" + sortField + "' does not exist in the base schema")
+          return false
+        }
+      }
+    } else {
+      // If no sort query is specified, then we will use one of the fields from the querySchema
+      val sortField = querySchema.fieldNames(0)
+      query.addSort(sortField, SolrQuery.ORDER.asc)
+      log.info("Added sort param '" + query.getSortField + "' to the query")
```

Comment:
this breaks the "functional" contract in that it's causing a side-effect on the query. Maybe move that code out of this method and only decide true or false in isStreamingPossible